critiquebrainz/db/review.py

```diff
@@ -0,0 +1,625 @@
+    # Create new revision
+    new_revision = db_revision.create(review_id, text)
+    review = get_by_id(review_id)
+    cache.invalidate_namespace(REVIEW_CACHE_NAMESPACE)
+    return review
+
+
+def create(*, release_group=None, entity_id=None,
+           entity_type=None, user_id, is_draft, text,
+           language=DEFAULT_LANG, license_id=DEFAULT_LICENSE_ID,
+           source=None, source_url=None):
+    """Create a new review on an entity.
+    Args:
+        release_group, entity_id (uuid): ID of the release group to be reviewed.
+        entity_id (str): Type of the entity reviewed.
+        user_id (uuid): ID of the reviewer.
+        is_draft (bool): Whether the review is drafted.
+        text (str): Content of the review.
+        language (str): Language of the review written. (default: 'en')
+        license_id (str): ID of the license.
+        source (str): Source of the review.
+        source_url (str): Url of the source of the review.
+
+    Returns:
+        Dictionary with the following structure
+        {
+            "id": uuid,
+            "entity_id": uuid,
+            "entity_type": str("release group", "event", "place"),
+            "user_id": uuid,
+            "user": dict,
+            "edits": int,
+            "is_draft": bool,
+            "is_hidden": bool,
+            "language": str,
+            "license_id": str,
+            "source": str,
+            "source_url": str,
+            "last_revision: dict,
+            "votes": dict,
+            "rating": int,
+            "text": str,
+            "created": datetime,
+            "license": dict,
+        }
+    """
+    if release_group:
+       entity_id = release_group
+       entity_type = "release_group"
+    else:
+        if not entity_type or not entity_id:
+          raise TypeError("Entity Type or ID not defined")
+
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            INSERT INTO review
+            VALUES (:id, :entity_id, :entity_type, :user_id, :edits, :is_draft,
+            :is_hidden, :license_id, :language, :source, :source_url)
+         RETURNING id;
+       """), {
+           "id": str(uuid.uuid4()),
+           "entity_id": entity_id,
+           "entity_type": entity_type,
+           "user_id": user_id,
+           "edits": 0,
+           "is_draft": is_draft,
+           "is_hidden": False,
+           "language": language,
+           "license_id": license_id,
+           "source": source,
+           "source_url": source_url,
+        })
+
+        review_id = result.fetchone()[0]
+        review_revision = db_revision.create(review_id, text)
+        cache.invalidate_namespace(REVIEW_CACHE_NAMESPACE)
+    return get_by_id(review_id)
+
+
+def list_reviews(*, inc_drafts=False, inc_hidden=False, entity_id=None,
+                 entity_type=None, license_id=None, user_id=None,
+                 language=None, exclude=None, sort=None, limit=None,
```

Comment:
It would be a good idea to provide a reasonable default `limit` to avoid issues in case it's not provided. 20 sounds like a good number.